### PR TITLE
Upgrade buildkit to current master

### PR DESCRIPTION
**Pull request by aaronlehmann**
Trying to keep up to date with all the API changes...

This isn't quite ready to yet. `TestYAML` sometimes fails (seems inconsistent):

```
--- FAIL: TestYAML (0.01s)
    --- FAIL: TestYAML/dockerfile (0.05s)
        yaml_test.go:215:
                Error Trace:    /Users/alehmann/src/llblib/yaml_test.go:215
                Error:          Not equal:
                                expected: "# [download 2/2] COPY --from=hi hi hi\n- type: FILE\n  actions:\n    - type: COPY\n      src: /hi\n      dest: /hi\n      follow-symlinks: true\n      contents-only: true\n      create-dest-path: true\n      allow-wildcard: true\n      allow-empty-wildcard: true\n      dest-input:\n        # [download 1/2] COPY --from=start start start\n        type: FILE\n        actions:\n            - type: COPY\n              src: /start\n              dest: /start\n              follow-symlinks: true\n              contents-only: true\n              create-dest-path: true\n              allow-wildcard: true\n              allow-empty-wildcard: true\n              dest-input:\n                type: SOURCE\n                source: scratch\n              src-input:\n                # [start 2/2] RUN echo start > start\n                type: EXEC\n                args: [/bin/sh, -c, echo start > start]\n                env: ['PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin']\n                cwd: /\n                mounts:\n                    - mountpoint: /\n                      type: BIND\n                      output: 0\n                      input: &ref0_0\n                        # [hi 1/2] FROM docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16\n                        type: SOURCE\n                        source: docker-image://docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16\n                        platform: linux/arm64\n      src-input:\n        # [hi 2/2] RUN echo hi > hi\n        type: EXEC\n        args: [/bin/sh, -c, echo hi > hi]\n        env: ['PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin']\n        cwd: /\n        mounts:\n            - mountpoint: /\n              type: BIND\n              output: 0\n              input: *ref0_0\n"
                                actual  : "# [download 2/2] COPY --from=hi hi hi\n- type: FILE\n  actions:\n    - type: COPY\n      src: /hi\n      dest: /hi\n      follow-symlinks: true\n      contents-only: true\n      create-dest-path: true\n      allow-wildcard: true\n      allow-empty-wildcard: true\n      dest-input:\n        # [download 1/2] COPY --from=start start start\n        type: FILE\n        actions:\n            - type: COPY\n              src: /start\n              dest: /start\n              follow-symlinks: true\n              contents-only: true\n              create-dest-path: true\n              allow-wildcard: true\n              allow-empty-wildcard: true\n              dest-input:\n                type: SOURCE\n                source: scratch\n              src-input:\n                # [start 2/2] RUN echo start > start\n                type: EXEC\n                args: [/bin/sh, -c, echo start > start]\n                env: ['PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin']\n                cwd: /\n                mounts:\n                    - mountpoint: /\n                      type: BIND\n                      output: 0\n                      input: &ref0_0\n                        # [start 1/2] FROM docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16\n                        type: SOURCE\n                        source: docker-image://docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16\n                        platform: linux/arm64\n      src-input:\n        # [hi 2/2] RUN echo hi > hi\n        type: EXEC\n        args: [/bin/sh, -c, echo hi > hi]\n        env: ['PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin']\n        cwd: /\n        mounts:\n            - mountpoint: /\n              type: BIND\n              output: 0\n              input: *ref0_0\n"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -37,3 +37,3 @@
                                                       input: &ref0_0
                                -                        # [hi 1/2] FROM docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16
                                +                        # [start 1/2] FROM docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16
                                                         type: SOURCE
                Test:           TestYAML/dockerfile
    yaml_test.go:20: build output:
        #1 [internal] load metadata for docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16
        #1 DONE 0.0s
FAIL
exit status 1
FAIL    github.com/coryb/llblib 1.848s
```

### CODE REVIEW

**Comment by coryb** [comment]
Yeah, TestYAML is known to be random.  The LLB generated from Dockerfile2LLB seems to generate random LLB in cases where ordering is not important.

**Review by coryb — APPROVED** [review]
lgtm.   I think I need to fix the TestYAML to allow for some pattern in the fixtures, or maybe I should generate multiple possible fixture files to test against and allow if any of them match.

File: solver.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ type Solver interface {
 	// domain socket in the build container.
 	Forward(src, dest string, opts ...ForwardOption) llb.RunOption
 	// Local will add a local directory to the solve request and session.
-	Local(name string, opts ...llb.LocalOption) llb.State
+	Local(name, path string, opts ...llb.LocalOption) llb.State
```
Comment (coryb) [inline solver.go]: Why this change?  it doesnt seem necessary.  I was trying to keep it consistent with the llb.Local api.

Comment (aaronlehmann) [reply solver.go]: The new `dockerfile2llb` code looks for a local named `context`, and it's hard to override that behavior.

Comment (coryb) [reply solver.go]: Hmm, maybe we can just create an llb.LocalOption that allows us to customize the name in the case for dockerfile2llb?  Then the 1st required argument is just `path`?

**Review by coryb — APPROVED** [review]
lgtm
